### PR TITLE
Working Track Selection

### PR DESCRIPTION
**Pull request by GN13008**
# Create Profil - select some track

## Description

This PR working better with PR from back.
You can now select song during profil selection.

## Todo

Save selection during the next step

### CODE REVIEW

File: expo-boomboomapp/src/components/SongPicker.tsx
Comment (cimandef) [inline expo-boomboomapp/src/components/SongPicker.tsx]: Here you could use guard clause : ) 

```suggestion
    if (!searchString || searchString.trim() === "") {
       return; // No need to proceed if the search string is empty
    }
    
    spotifyApiService
      .fetchTracksNyName(searchString)
      .then((tracks) => {
         setFetchedSongs(tracks.data || []);
      })
      .catch((error) => {
          // TODO: handle error better
          console.error("spotifyApiService:", error);
      });
```

You can read more about this 'guard clause' [here](https://www.gavsblog.com/blog/how-to-and-why-use-guard-clauses-in-javascript). 
It's just a tiny suggestion, not a blocker to merge this PR !

File: expo-boomboomapp/src/components/SongPicker.tsx
Comment (cimandef) [inline expo-boomboomapp/src/components/SongPicker.tsx]: Don't remember to remove your `console.log` before merge ;)

**Review by cimandef — APPROVED** [review]
All good on my side, just a tiny suggestion but not a blocker ! Let's merge it if @llucasspot  It's okay !

File: expo-boomboomapp/src/components/SongPicker.tsx
Comment (llucasspot) [inline expo-boomboomapp/src/components/SongPicker.tsx]: be carefull

File: expo-boomboomapp/src/api/SpotifyApiService/SpotifyApiServiceI.ts
Comment (llucasspot) [inline expo-boomboomapp/src/api/SpotifyApiService/SpotifyApiServiceI.ts]: I think it's bad to adapt the frontend according to the backend has it is now, here by exemple the `status` field is not needed because the info is already in the http response. 

If the response is 200 we know already if the backend is good or not. We need to play with response http status and not with respondy body status. 

I will rework the backend. In the meantime we can develop frontend separately from the backend with mocks class. And at the end we will merge the two part. 
What do you think ?

**Comment by llucasspot** [comment]
Hi, 
To follow up on my comment above I will not approve this PR.
I think the best way to develop the app is to firstly develop the frontend with just the info that it needs with api mocks.

At the same time, we will develop backend and the needed endpoints.
And at the end we will just need to update api type and endpoint url when backend will be ready.
What do you think @Doelia ?

File: expo-boomboomapp/src/components/SongPicker.tsx
Comment (GN13008) [reply expo-boomboomapp/src/components/SongPicker.tsx]: About ?

File: expo-boomboomapp/src/components/SongPicker.tsx
Comment (GN13008) [reply expo-boomboomapp/src/components/SongPicker.tsx]: Yes it's a nice comment i will push changes

File: expo-boomboomapp/src/api/SpotifyApiService/SpotifyApiServiceI.ts
Comment (GN13008) [reply expo-boomboomapp/src/api/SpotifyApiService/SpotifyApiServiceI.ts]: I think it's good to have a body with data and other things, it's more scalable. But yes the status can be manage differently

**Comment by GN13008** [comment]
Did you see my PR on backend ? 

I like to dev with real data this is why I work on back end front simultaneously. I will improve my code like Angel suggest waiting to your decision ...

File: expo-boomboomapp/src/components/SongPicker.tsx
Comment (llucasspot) [reply expo-boomboomapp/src/components/SongPicker.tsx]: you have write setFetchedSongs([]); and setFetchedSongs(tracks.data); next to each other

**Review by llucasspot — APPROVED** [review]

**Comment by llucasspot** [comment]
Hi, 
Sorry I did not see your PR on backend. I understand now the changes that you have made. It is merger ;)

Be carefull when you change a class Interface like `SpotifyApiServiceI` to change also the mock class like `SpotifyApiMockService` to prevent bugs if others dev work with the `MOCK_MODE`. 
see : https://github.com/llucasspot/boomboom-front-react-native/pull/12/commits/7568f01c0604a80bdbe0f360901461e08b3cd920

Otherwise good job ;)

**Pull request merged**
